### ISSUE TITLE
Case splitting on infix copattern generates invalid syntax

### ISSUE THREAD

**Issue by m0davis** [agda/agda#2332]
```agda
record R : Set₁ 
 where
  field
    _f_ : Set → Set
```
Start with something like this...
```agda
instance I₁ : R
I₁ = {!!} 
```
...and after case splitting (C-c C-c), we end up with something like this ...
```agda
instance I₂ : R
R._f_ I₂ = {!!}
```
That's fine, but case split again and we end up something Agda doesn't know how to parse:
```agda
instance I₃ : R
I₃ R.f x = ?
{-
Don't know how to parse I₃ R.f x. Could mean any one of:
  (I₃ R.f) x
  R._f_ I₃ x
Operators used in the grammar:
  R.f (infix operator, level 20)
-}
```
I would have hoped for something like this when case-splitting on `I₂`:
```agda
instance I₄ : R
R._f_ I₄ x = {!!}
```

**Comment by andreasabel** [comment]
Fixing #632 would fix this issue. 

Side remark: I am a bit surprised that Agda suggests `(I₃ R.f) x` as a possible parse? @nad, do you know why?

**Comment by nad** [comment]
If you compile using `make install-debug` and then run `agda-<version>-debug -vscope.grammar:10 <file>`, then you can see the grammar that is used:
```
<TopK>
  where
  <AppK> ∷= <NonfixK>
            ((<NonfixK> >>= ?) | <implicit argument> | <instance argument>) ⋆
  <NonfixK> ∷= <atom> | ∅
  <TopK> ∷= <NodeK (Right 20)> | ∅
  <NodeK (Right 20)> ∷= <AppK> | <AppK> ("R.f" ε | ∅) <AppK> | ∅
```
I guess that when parsing this pattern `R.f` is a valid atom, see [`isAtom`](https://github.com/agda/agda/blob/61da63a75153384e9dc3ff329b62e360e0206c73/src/full/Agda/Syntax/Concrete/Operators.hs#L279-L281) and the documentation of `buildParsers`:

> When parsing a pattern we do not use bound names. The effect is that operator parts (that are not constructor parts) can be used as atomic names in the pattern (so they can be rebound). See @test/succeed/OpBind.agda@ for an example.

However, I suspect that this feature was not intended to apply to *qualified* operator parts.

**Comment by andreasabel** [comment]
I did not know there were debug versions of Agda.  Actually, the Makefile does not seem to know it either.
```
$ make install-debug
make: *** No rule to make target 'install-debug'.  Stop.
```

**Comment by gallais** [comment]
I think that's the thing: https://github.com/agda/agda/commit/3162e0c519dc25bf1b43f12661d6a4a23b38c7d0

**Comment by andreasabel** [comment]
A thanks, it did not sound like a brand new thing...

**Comment by UlfNorell** [comment]
`I₁ R.f x` parses as expected now.

**Comment by UlfNorell** [comment]
Looks like it was fixed in `2.5.2`.
